homeassistant/config_entries.py

```diff
@@ -2376,6 +2376,20 @@ def __init_subclass__(cls, *, domain: str | None = None, **kwargs: Any) -> None:
         if domain is not None:
             HANDLERS.register(domain)(cls)
 
+    @property
+    def _reauth_entry_id(self) -> str | None:
+        """Return reauth entry id."""
+        if self.source == SOURCE_REAUTH:
+            return self.context.get("entry_id")
+        return None
```

Comment:
I think we should raise in this case

Reply:
Now raising `ValueError`

Reply:
Morning thought... maybe AttributeError would be better